processing/src/main/java/io/druid/query/groupby/epinephelinae/GroupByMergingQueryRunnerV2.java

```diff
@@ -0,0 +1,661 @@
+                      Lists.newArrayList(
+                          Iterables.transform(
+                              queryables,
+                              new Function<QueryRunner<Row>, ListenableFuture<Void>>()
+                              {
+                                @Override
+                                public ListenableFuture<Void> apply(final QueryRunner<Row> input)
+                                {
+                                  if (input == null) {
+                                    throw new ISE("Null queryRunner! Looks to be some segment unmapping action happening");
+                                  }
+
+                                  return exec.submit(
+                                      new AbstractPrioritizedCallable<Void>(priority)
+                                      {
+                                        @Override
+                                        public Void call() throws Exception
+                                        {
+                                          try {
+                                            input.run(queryForRunners, responseContext).accumulate(grouper, accumulator);
+                                            return null;
+                                          }
+                                          catch (QueryInterruptedException e) {
+                                            throw Throwables.propagate(e);
+                                          }
+                                          catch (Exception e) {
+                                            log.error(e, "Exception with one of the sequences!");
+                                            throw Throwables.propagate(e);
+                                          }
+                                        }
+                                      }
+                                  );
+                                }
+                              }
+                          )
+                      )
+                  );
+
+                  waitForFutureCompletion(query, futures, timeoutAt - processingStartTime);
+
+                  return new CloseableGrouperIterator<>(
+                      grouper,
+                      true,
+                      new Function<Grouper.Entry<GroupByMergingKey>, Row>()
+                      {
+                        @Override
+                        public Row apply(Grouper.Entry<GroupByMergingKey> entry)
+                        {
+                          Map<String, Object> theMap = Maps.newLinkedHashMap();
+
+                          // Add dimensions.
+                          for (int i = 0; i < entry.getKey().getDimensions().length; i++) {
+                            theMap.put(
+                                query.getDimensions().get(i).getOutputName(),
+                                Strings.emptyToNull(entry.getKey().getDimensions()[i])
```

Comment:
IncrementalIndex would have returned this as empty string for nulls,
are you sure we need to convert empty strings back to null here ?